ui/app/pages/permissions-connect/permissions-connect.container.js

```diff
@@ -0,0 +1,57 @@
+import { connect } from 'react-redux'
+import PermissionApproval from './permissions-connect.component'
+import {
+  accountsWithSendEtherInfoSelector,
+  getFirstPermissionRequest,
+  getNativeCurrency,
+} from '../../selectors/selectors'
+import { approvePermissionsRequest, rejectPermissionsRequest, showModal, getOpenMetaMaskTabs, getCurrentWindowTab } from '../../store/actions'
+
+const mapStateToProps = state => {
+  const permissionsRequest = getFirstPermissionRequest(state)
+  const { metadata = {} } = permissionsRequest || {}
+  const { origin } = metadata
+  const nativeCurrency = getNativeCurrency(state)
+
+  const accountsWithoutLabel = accountsWithSendEtherInfoSelector(state)
+  const accountsWithLabel = accountsWithoutLabel.map(account => {
```

Comment:
This code for getting `accountsWithLabel` is in two places. It can be put into a selector.

Reply:
30a7b3738